### ISSUE TITLE
Fresh install stops at running kasm_db container

### ISSUE THREAD

**Issue by pdav0** [kasmtech/workspaces-issues#150]
Hello,

I'm trying to install Kasm 1.10 on a freshly installed Ubuntu server 20.04.3 LTS on a VirtualBox VM. The install aborts with the following messages:

Initializing Database
Setting Seed file as /opt/kasm/1.10.0/conf/database/seed_data/default_properties.yaml
[+] Running 1/1
 ⠿ Container kasm_db  Stopped                                              3.1s
Error response from daemon: Cannot kill container: kasm_db: Container 608e07d3aaa83ece80f10ef57e4519d7369a312354b54a017d5fec7d994e1bcc is not running
kasm_db
kasm_db_1.10.0
kasm_db_1.10.0
[+] Running 1/1
 ⠿ Container kasm_db  Started                                              3.5s
2022-02-07 14:29:55,950 [INFO] root: Performing Database Connectivity Test
2022-02-07 14:29:55,991 [INFO] root: Initializing Postgres Database
Traceback (most recent call last):
  File "sqlalchemy/engine/base.py", line 2262, in _wrap_pool_connect
  File "sqlalchemy/pool/base.py", line 363, in connect
  File "sqlalchemy/pool/base.py", line 760, in _checkout
  File "sqlalchemy/pool/base.py", line 492, in checkout
  File "sqlalchemy/pool/impl.py", line 139, in _do_get
  File "sqlalchemy/util/langhelpers.py", line 68, in __exit__
  File "sqlalchemy/util/compat.py", line 129, in reraise
  File "sqlalchemy/pool/impl.py", line 136, in _do_get
  File "sqlalchemy/pool/base.py", line 308, in _create_connection
  File "sqlalchemy/pool/base.py", line 437, in __init__
  File "sqlalchemy/pool/base.py", line 639, in __connect
  File "sqlalchemy/engine/strategies.py", line 114, in connect
  File "sqlalchemy/engine/default.py", line 453, in connect
  File "psycopg2/__init__.py", line 126, in connect
psycopg2.OperationalError: could not connect to server: Connection refused
	Is the server running on host "db" (172.18.0.2) and accepting
	TCP/IP connections on port 5432?

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "api_server/server.py", line 100, in <module>
  File "initialize_postgres_db.py", line 16, in __init__
  File "data/data_access_factory.py", line 12, in createSession
  File "data/access_postgres.py", line 42, in __init__
  File "sqlalchemy/dialects/postgresql/base.py", line 2557, in has_table
  File "sqlalchemy/engine/base.py", line 2165, in execute
  File "sqlalchemy/engine/base.py", line 2226, in _contextual_connect
  File "sqlalchemy/engine/base.py", line 2266, in _wrap_pool_connect
  File "sqlalchemy/engine/base.py", line 1536, in _handle_dbapi_exception_noconnection
  File "sqlalchemy/util/compat.py", line 383, in raise_from_cause
  File "sqlalchemy/util/compat.py", line 128, in reraise
  File "sqlalchemy/engine/base.py", line 2262, in _wrap_pool_connect
  File "sqlalchemy/pool/base.py", line 363, in connect
  File "sqlalchemy/pool/base.py", line 760, in _checkout
  File "sqlalchemy/pool/base.py", line 492, in checkout
  File "sqlalchemy/pool/impl.py", line 139, in _do_get
  File "sqlalchemy/util/langhelpers.py", line 68, in __exit__
  File "sqlalchemy/util/compat.py", line 129, in reraise
  File "sqlalchemy/pool/impl.py", line 136, in _do_get
  File "sqlalchemy/pool/base.py", line 308, in _create_connection
  File "sqlalchemy/pool/base.py", line 437, in __init__
  File "sqlalchemy/pool/base.py", line 639, in __connect
  File "sqlalchemy/engine/strategies.py", line 114, in connect
  File "sqlalchemy/engine/default.py", line 453, in connect
  File "psycopg2/__init__.py", line 126, in connect
sqlalchemy.exc.OperationalError: (psycopg2.OperationalError) could not connect to server: Connection refused
	Is the server running on host "db" (172.18.0.2) and accepting
	TCP/IP connections on port 5432?

(Background on this error at: http://sqlalche.me/e/e3q8)
[8] Failed to execute script 'server' due to unhandled exception!

I appreciate any help you can give me.

Best Regards,

Philippe David

**Comment by j-travis** [comment]
Hi, this has been posted by other users. Can you review the existing tickets:
https://github.com/kasmtech/workspaces-issues/issues/99
https://github.com/kasmtech/workspaces-issues/issues/80
https://github.com/kasmtech/workspaces-issues/issues/16

**Comment by pdav0** [comment]
Hello,

I configured my firewall to open ports 443, 5432 and 6379 in the trusted zone.
I also set IPs 172.18.0.0/24 in the trusted zone and also the interface (br-ec18a4fa0402) in the trusted zone. Nothing worked. Is it possible that the fact that my server is running in a VM could be putting a spanner in the works? I set the network connection in my VM to bridged.

**Comment by j-travis** [comment]
Once the installer fails can your send the output of the following commands

```
sudo docker ps
```
```
sudo docker logs kasm_db
```

Then try and launch a separate container and try to connect the database via telnet. This will tell you if there are any firewalls blocking the connection. Use the IP and port from the error log when the install fails:

> Is the server running on host "db" (172.18.0.2) and accepting
> TCP/IP connections on port 5432?


```
sudo docker run -it --rm --network kasm_default_network busybox sh
# telnet 172.18.0.2 5432
Connected to 172.18.0.2
```

If it works you should get the Connected statement

**Comment by pdav0** [comment]
Hello,

here's the result of what you suggested:

neo@bdvmkasm:/tmp$ sudo docker ps
CONTAINER ID   IMAGE                COMMAND                  CREATED          STATUS          PORTS      NAMES
4756c002a734   postgres:12-alpine   "docker-entrypoint.s…"   52 seconds ago   Up 48 seconds   5432/tcp   kasm_db
neo@bdvmkasm:/tmp$ sudo docker logs kasm_db
The files belonging to this database system will be owned by user "postgres".
This user must also own the server process.

The database cluster will be initialized with locale "en_US.utf8".
The default database encoding has accordingly been set to "UTF8".
The default text search configuration will be set to "english".

Data page checksums are disabled.

fixing permissions on existing directory /var/lib/postgresql/data ... ok
creating subdirectories ... ok
selecting dynamic shared memory implementation ... posix
selecting default max_connections ... 100
selecting default shared_buffers ... 128MB
selecting default time zone ... UTC
creating configuration files ... ok
running bootstrap script ... ok
sh: locale: not found
2022-02-10 13:12:19.379 UTC [31] WARNING:  no usable system locales were found
performing post-bootstrap initialization ... ok
initdb: warning: enabling "trust" authentication for local connections
You can change this by editing pg_hba.conf or using the option -A, or
--auth-local and --auth-host, the next time you run initdb.
syncing data to disk ... ok


Success. You can now start the database server using:

    pg_ctl -D /var/lib/postgresql/data -l logfile start

waiting for server to start....< 2022-02-10 13:13:15.724 UTC     37  00000 2022-02-10 13:13:15 UTC 62050f6b.25:>LOG:  starting PostgreSQL 12.9 on x86_64-pc-linux-musl, compiled by gcc (Alpine 10.3.1_git20211027) 10.3.1 20211027, 64-bit
< 2022-02-10 13:13:15.820 UTC     37  00000 2022-02-10 13:13:15 UTC 62050f6b.25:>LOG:  listening on Unix socket "/var/run/postgresql/.s.PGSQL.5432"
< 2022-02-10 13:13:16.112 UTC     37  00000 2022-02-10 13:13:15 UTC 62050f6b.25:>LOG:  redirecting log output to logging collector process
< 2022-02-10 13:13:16.112 UTC     37  00000 2022-02-10 13:13:15 UTC 62050f6b.25:>HINT:  Future log output will appear in directory "/var/log/postgres".
. done
server started
neo@bdvmkasm:/tmp$ sudo docker run -it --rm --network kasm_default_network busybox sh
Unable to find image 'busybox:latest' locally
latest: Pulling from library/busybox
009932687766: Pull complete 
Digest: sha256:afcc7f1ac1b49db317a7196c902e61c6c3c4607d63599ee1a82d702d249a0ccb
Status: Downloaded newer image for busybox:latest
/ # telnet 172.18.0.2 5432
Connected to 172.18.0.2

Looks like it works. I guess I need to recheck my firewall settings again or my VM settings.

**Comment by j-travis** [comment]
If you still continue to have issues I have one more thing you can try.
During the install steps, after you extract the package but before you call `install.sh` edit the following file

```
/tmp/kasm_release/bin/utils/db_init
```

Toward the end of the file there is a `sleep 5` . Change that to `sleep 30` then run the normal `install.sh` .

This logic should really be improved on our end, but its a simple pause to wait until the database container  _should_ be started and working. Maybe on your system it starts slower than expected so the following steps fail.

**Comment by dc-bond** [comment]
Hi, I just wanted to drop in to say that I was having the exact same issue as above, and I had to change the sleep setting to a full 90 seconds to get this to work. It seems like my VM is just really slow at booting up the database. Thanks!

**Comment by j-travis** [comment]
This was addressed in 1.11.0. The installer now properly waits for the DB container to reach a healthy state before trying to access it
